chalk-solve/src/rust_ir.rs

```diff
@@ -168,6 +168,15 @@ pub struct FnDefDatumBound<I: Interner> {
     pub where_clauses: Vec<QuantifiedWhereClause<I>>,
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct ClosureDatum<I: Interner> {
```

Comment:
Doc comment, also for the fields -- it's nice to have something show up in rustdoc :)